src/util.cpp

```diff
@@ -375,6 +376,27 @@ int LogPrintStr(const std::string &str)
     return ret;
 }
 
+bool LockDirectory(const fs::path& directory, const std::string lockfile_name, bool probe_only)
```

Comment:
Maybe combine `directory` and `std::string lockfile_name` arguments into a single `path` argument. Having 2 arguments seems more verbose and less flexible.

Reply:
Indeed, was just for the purpose of printing out the directory in the log message below rather than the filename too. Will modify if/when there's another change too

Reply:
I kind of like these separated out - after all, the function is called `LockDirectory` and not `CreateLockFile`. If you merge the directory and lockfile name, I'd say you'd also have to rename the function because it's not longer specifically about locking directories.

Reply:
Nit: `const std::string &lockfile_name`